src/integrationTest/java/com/example/javafxtest/integrationtest/FileChooserApplicationTest.java

```diff
@@ -0,0 +1,167 @@
+package com.example.javafxtest.integrationtest;
+
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.DialogPane;
+import javafx.scene.control.Label;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testfx.api.FxRobot;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.ApplicationTest;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.testfx.assertions.api.Assertions.assertThat;
+
+import com.example.javafxtest.FileChooserApplication;
+
+// https://github.com/Oliver-Loeffler/FXFileChooser/blob/master/src/test/java/net/raumzeitfalle/fx/filechooser/FileChooserControllerTest.java
+
+@ExtendWith(ApplicationExtension.class)
+public class FileChooserApplicationTest { // extends AbstractApplicationTest {
+
+    private static final String LAME_EXCUSE = "It seems file choosers are rooted in the concrete OS and can't be controlled nicely using TestFX. " +
+            "Let's trust the framework, and, in another test, inject member values so we can test other things.";
+    private Button openCsvButton;
+    private Label csvFileLabel;
+    private Button openTexButton;
+    private Label texFileLabel;
+    private Button renderPdfButton;
+
+    @BeforeAll
+    public static void setupForHeadlessTesting() {
+        HeadlessHelper.setupForHeadlessTesting();
+    }
+
+    @BeforeEach
+    public void setup() throws Exception {
+        ApplicationTest.launch(FileChooserApplication.class);
+    }
+
+    /**
+     * @param robot - Will be injected by the test runner.
+     */
+    @Test
+    public void testInitialUiTexts(FxRobot robot) {
+        // arrange
+        lookUpUiNodes(robot);
+
+        // assert
+        assertThat(openCsvButton).hasText("Pick a CSV file!");
+        assertThat(csvFileLabel).hasText("[CSV file]");
+        assertThat(openTexButton).hasText("Pick a TEX file!");
+        assertThat(texFileLabel).hasText("[TEX file]");
+        assertThat(renderPdfButton).hasText("Render PDFs!");
+    }
+
+
+    @Test
+    @Disabled(LAME_EXCUSE)
+    // see GtkCommonDialogs: private static native FileChooserResult _showFileChooser
+    void testFileSelection(FxRobot robot) {
+        // arrange
+        lookUpUiNodes(robot);
+        File dummyCsv = getDummyFileFromClasspath("dummyCsv.csv");
+        File dummyTex = getDummyFileFromClasspath("dummyTex.tex");
+
+        // act
+        robot.clickOn(openCsvButton);
+
+        // assert
+        assertThat(openCsvButton).hasText("Pick a CSV file!");
+        assertThat(csvFileLabel).hasText(dummyCsv.getAbsolutePath());
+        assertThat(csvFileLabel.getTooltip().getText()).isEqualTo(dummyCsv.getAbsolutePath());
+    }
+
+    @Test
+    @Disabled("not yet fully implemented")
```

Comment:
- [ ] Implement proper tests (find a way to inject a controller or re-design the tests)